cumulus/pallets/xcmp-queue/src/lib.rs

```diff
@@ -649,11 +647,10 @@ impl<T: Config> Pallet<T> {
 			drop_threshold,
 		);
 
-		// `batches_footprints[n]` contains the footprint of the batch `xcms[0..n]`,
-		// so as `n` increases `batches_footprints[n]` contains the footprint of a bigger batch.
-		let best_batch_idx = batches_footprints.binary_search_by(|batch_info| {
+		let best_batch_footprint = batches_footprints.search_best_by(|batch_info| {
 			let required_weight = T::WeightInfo::enqueue_xcmp_messages(
 				batch_info.new_pages_count,
+				batches_footprints.first_page_pos.saturated_into(),
 				batch_info.msgs_count,
 				batch_info.size_in_bytes,
 			);
```

Comment:
Wouldn't it make sense to pass the whole `BatchFootprint` here?

```suggestion
			let required_weight = T::WeightInfo::enqueue_xcmp_messages(
				batch_info,
				batches_footprints.first_page_pos.saturated_into(),
			);
```

Reply:
Yes, looks good. Done.